src/workerd/api/tests/http-socket-test.js

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2017-2024 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+import { connect, createHttpClient } from 'cloudflare:sockets';
+import { strict as assert } from 'node:assert';
+
+export default {
+  async test(ctrl, env) {
+    // Basic connectivity and GET test
+    {
+      const socket = connect(`localhost:${env.HTTP_SOCKET_SERVER_PORT}`);
+      const httpClient = createHttpClient(socket);
```

Comment:
What happens if I do `socket.readable.getReader()` after this point....

Reply:
Hell breaks loose.

Reply:
`ETOOMUCHCHAOS`